File: Leetcode 2785. Sort Vowels in a String/lc 2785.cpp
class Solution {
public:
    char vowels[5]={'a','e','i','o','u'};
    int find(char c){
        for(int i=0;i<5;i++){
            if(c==vowels[i]){
                return 1;
            }
        }
        return 0;
    }
    string sortVowels(string s) {
        vector<char> vowels_found;
        
        for(int i=0;i<s.length();i++){
            char c=tolower(s[i]);
            if(find(c)) vowels_found.emplace_back(s[i]);      
        }

        sort(vowels_found.begin(),vowels_found.end());
        int ind=0;
        for(int i=0;i<s.length();i++){
            if(find(tolower(s[i]))){
                s[i]=vowels_found[ind++];
            }
        }
        return s;
        
    }
};

File: Leetcode 2785. Sort Vowels in a String/lc 2785.java
class Solution {
    public String sortVowels(String s) {
        char[] ch =s.toCharArray();
        String ans="";
        ArrayList<Character> vow=new ArrayList<Character>();
        for(int i=0;i<ch.length;i++){
            if(check(ch[i])){
                vow.add(ch[i]);
                ch[i]='u';
            }
        }
        Collections.sort(vow);
        int c=0;
        for(int i=0;i<ch.length;i++){
            if(ch[i]=='u'){
                ans+=String.valueOf(vow.get(c));
                c++;
            }else{
                ans+=String.valueOf(ch[i]);
            }
            
        }
        return ans;
    }
    
    boolean check(char c){
        char[] vowels={'A','E','I','O','U','a','e','i','o','u'};
        for(int i=0;i<vowels.length;i++){
            if(c==vowels[i]){
                return true;
            }
        }
        return false;
    }
}